Day-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,146 @@
+import "./App.css";
+import Display from "./components/DisplayComponents/CalculatorDisplay";
+import WhiteButtons from "./components/ButtonComponents/NumberButton";
+import RedButtons from "./components/ButtonComponents/ActionButton";
 
+const funcs = ["÷", "x", "-", "+", "="];
+const ints = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      displayVal: "0",
+      postOp: false,
+      storedVal: null,
+      storedOp: null
+    };
+  }
+
+  clear() {
+    let disp = this.state.displayVal;
+    if (this.state.postOp) {
+      this.setState({ displayVal: "confirm clear" });
+    }
+    if (!this.state.postOp || disp === "confirm clear") {
+      this.setState({
+        displayVal: "0",
+        postOp: false,
+        postEqual: false,
+        storedVal: null,
+        storedOp: null
+      });
+    }
+  }
+
+  numInput(num) {
+    let disp = this.state.displayVal;
+    if (disp.length > 11) {
+      this.setState({ displayVal: "it done broke" });
+      return;
+    }
+    if (disp === "0" || disp === this.state.storedVal || this.state.postEqual || disp === "confirm clear") {
+      this.setState({ displayVal: String(num), postEqual:false});
+    } else {
+      this.setState({ displayVal: disp + num });
+    }
+  }
+
+  opInput(func) {
+    let disp = this.state.displayVal;
+
+    if (func === "÷") {
+      this.setState({
+        postOp: true,
+        storedVal: disp,
+        storedOp: func
+      });
+    }
+    if (func === "x") {
+      this.setState({
+        postOp: true,
+        storedVal: disp,
+        storedOp: func
+      });
+    }
+    if (func === "-") {
+      this.setState({
+        postOp: true,
+        storedVal: disp,
+        storedOp: func
+      });
+    }
+    if (func === "+") {
+      this.setState({
+        postOp: true,
+        storedVal: disp,
+        storedOp: func
+      });
+    }
+    if (func === "=") {
+      let stored = this.state.storedVal;
+      let op = this.state.storedOp;
+      let result;
+      if (op === "÷") {
+        result = parseFloat(stored) / parseFloat(disp);
+        result = result.toFixed(2);
+      }
+      if (op === "+") {
+        result = parseFloat(stored) + parseFloat(disp);
+      }
+      if (op === "-"){
+        result = parseFloat(stored) - parseFloat(disp);
+      }
+      if (op === "x"){
+        result = parseFloat(stored) * parseFloat(disp);
+      }
+      
+      this.setState({
+        displayVal: result,
+        postOp: false,
+        postEqual: true,
+        storedVal: null,
+        storedOp: null
+      });
+    }
+  }
+
+  render() {
+    return (
+      <div className="cal-container">
+        <Display value={this.state.displayVal} />
+        <div className="btn-container">
+          <div className="white-btns">
+            <div className="btn zero" onClick={() => this.numInput(0)}>
+              0
+            </div>
+            {ints.map(num => {
```

Comment:
Good use of map to generate all the buttons. Here is a cool trick to generate an array of integers `Array(9).fill(0).map((_, i) => i + 1)`. I'm gonna let you figure out how this works.